src/rttest.cpp

```diff
@@ -728,7 +781,7 @@ std::string rttest_results_to_string(struct rttest_results * results, char * nam
   sstring << "    - Standard deviation: " << results->latency_stddev << std::endl;
   sstring << std::endl;
 
-  return sstring.str();
+  return sstring.str().c_str();
```

Comment:
The succeeding test doesn't matter. It is still accessing already freed memory. Have you tried to run it with e.g. valgrind?